### PR TITLE
Add keadm compatible e2e test

### PR DESCRIPTION
**Pull request by wlq1212**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Add keadm compatibility e2e test

**Which issue(s) this PR fixes**:

Fixes #https://github.com/kubeedge/kubeedge/issues/4896

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -449,3 +449,20 @@ else
 release:
 	hack/make-rules/release.sh $(WHAT) $(ARM_VERSION) $(OS)
 endif
+
+define KEADM_COMPATIBILITY_E2E_HELP_INFO
+# eadm e2e test.
```
Comment (Rei1010) [inline Makefile:454]: keadm

Comment (wlq1212) [reply Makefile]: done

File: tests/scripts/keadm_compatibility_e2e.sh [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+   set +e
+   docker rmi $(docker images -f "dangling=true" -q)
+   docker system prune -f
+   set -Ee
+
+}
+
+function get_edgecore_image() {
+   docker run --rm kubeedge/installation-package:$CLOUD_EDGE_VERSION cat /usr/local/bin/keadm > /usr/local/bin/edgeadm && chmod +x /usr/local/bin/edgeadm
```
Comment (ZhengXinwei-F) [inline tests/scripts/keadm_compatibility_e2e.sh:75]: Could you please assist me with understanding this code? Why should you use edgeadm join instead than keadm join?

Comment (wlq1212) [reply tests/scripts/keadm_compatibility_e2e.sh]: done

**Comment by wlq1212** [comment]
image

File: tests/scripts/keadm_compatibility_e2e.sh [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  kind delete cluster --name test
+  sudo rm -rf /var/log/kubeedge /etc/kubeedge /etc/systemd/system/edgecore.service $E2E_DIR/e2e_keadm/e2e_keadm.test $E2E_DIR/config.json
+}
+
+function build_ginkgo() {
+  cd $E2E_DIR
+  ginkgo build -r e2e_keadm/
+}
+
+function prepare_cluster() {
```
[the remark this answers is not in the document]
Comment (wlq1212) [reply tests/scripts/keadm_compatibility_e2e.sh:38]: There is a corresponding PRs, and I will modify it after the pr is merged.
link: https://github.com/kubeedge/kubeedge/pull/5283

**Comment by ZhengXinwei-F** [comment]
lgtm. One of my reviews appears to be missing.  :-)

File: tests/scripts/keadm_compatibility_e2e.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (ZhengXinwei-F) [reply tests/scripts/keadm_compatibility_e2e.sh:38]: ok

**Comment by wlq1212** [comment]
cc  @fisherxu @Shelley-BaoYue @WillardHu

**Comment by fisherxu** [comment]
Good job @wlq1212 :)  Will have a review soon

**Review by Shelley-BaoYue — COMMENTED** [review]
/lgtm

**Review by fisherxu — COMMENTED** [review]
/approve

**Pull request merged**